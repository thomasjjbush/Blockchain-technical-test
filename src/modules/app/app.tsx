import React, { FC, lazy, ReactElement, Suspense } from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
import { createGlobalStyle, ThemeProvider } from 'styled-components';
import { store } from '../../store/store';
import { Nav } from '../nav/nav';
import { StyledProps } from '../../../types';

const GlobalStyles = createGlobalStyle<StyledProps>`
    body {
        background-color: ${(props): string => props.theme.colors.offset};
        font-family: Sans-serif;
        margin: 0;

        * {
            box-sizing: border-box;
        }
    }
`;

const routes = [
    { component: lazy(() => import('./../address/address')), path: '/address/:slug' },
    { component: lazy(() => import('./../home/home')), path: '/' },
];

const theme = {
    colors: {
        brand: '#0C6CF2',
        offset: '#C0C0C0',
    },
};

export const App: FC = (): ReactElement => (
    <Provider store={store}>
        <ThemeProvider theme={theme}>
            <GlobalStyles />
            <Router>
                <Nav />
                <Suspense fallback={<div>Loading!</div>}>
                    <Switch>
                        {routes.map(({ component: Component, path }) => (
                            <Route exact key={path} path={path}>
                                <Component />
                            </Route>
                        ))}
                    </Switch>
                </Suspense>
            </Router>
        </ThemeProvider>
    </Provider>
);
