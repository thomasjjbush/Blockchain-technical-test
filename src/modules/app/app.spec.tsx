import React from 'react';
import { shallow } from 'enzyme';
import { App } from './app';

jest.mock('react', () => ({ ...jest.requireActual('react'), lazy: jest.fn(() => 'Component') }));

describe('App', () => {
    it('should match snapshot', () => {
        expect(shallow(<App />)).toMatchSnapshot();
    });
});
