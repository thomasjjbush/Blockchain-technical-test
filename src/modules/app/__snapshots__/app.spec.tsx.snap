// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Component
    theme={
      Object {
        "colors": Object {
          "brand": "#0C6CF2",
          "offset": "#C0C0C0",
        },
      }
    }
  >
    <Memo(l) />
    <BrowserRouter>
      <Nav />
      <Suspense
        fallback={
          <div>
            Loading!
          </div>
        }
      >
        <Switch>
          <Route
            exact={true}
            key="/address/:slug"
            path="/address/:slug"
          >
            <Component />
          </Route>
          <Route
            exact={true}
            key="/"
            path="/"
          >
            <Component />
          </Route>
        </Switch>
      </Suspense>
    </BrowserRouter>
  </Component>
</Provider>
`;
